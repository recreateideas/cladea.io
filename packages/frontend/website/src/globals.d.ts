import { Theme } from "./App/ui-core/theme";
import { Store } from "redux";

export declare global {
  interface Window {
    __REDUX_DEVTOOLS_EXTENSION_COMPOSE__?: any;
    /** enzyme global methods for testing */
    shallow: any;
    render: any;
    mount: any;
    theme: Theme;
    ResizeObserver: any;
    store: Store;
  }
}

declare module "ui-core";
