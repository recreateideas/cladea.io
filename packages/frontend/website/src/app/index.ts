export * from "./App";
export * from "./features";
