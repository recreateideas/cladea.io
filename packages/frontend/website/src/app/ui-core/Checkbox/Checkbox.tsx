import { forwardRef, ReactElement, Ref } from 'react';
import { FormControlLabel, Checkbox as CheckboxCore, CheckboxProps } from '@mui/material';
import { Container } from './styledComponents';

interface IProps extends CheckboxProps {
    label?: string | ReactElement;
}
export const Checkbox = forwardRef((props: IProps, ref: Ref<any>): ReactElement => {
    const { label, checked, value } = props;
    const isChecked = !!checked || !!value;
    const Input = <CheckboxCore ref={ref} checked={isChecked} {...props} />;
    return (
        <Container>{label ? <FormControlLabel label={label} control={Input} /> : Input}</Container>
    );
});

Checkbox.displayName = 'Checkbox';

Checkbox.defaultProps = {};
