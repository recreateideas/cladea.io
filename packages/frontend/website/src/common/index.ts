export * from './utils';
export * from './hooks';
export * from './converters';
