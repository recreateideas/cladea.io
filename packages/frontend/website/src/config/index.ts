export const apiGatewaUrl = process.env.REACT_APP_API_GATEWAY_URL;

console.log(process.env);
