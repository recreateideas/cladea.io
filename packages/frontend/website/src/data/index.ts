import land110m from "./land-110m.json";

export { land110m };
