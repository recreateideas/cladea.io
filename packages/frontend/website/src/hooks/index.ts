export * from "./ClickOutside";
export * from "./use-scroll-y";
export * from "./use-margin";
export * from "./use-breakpoints";
