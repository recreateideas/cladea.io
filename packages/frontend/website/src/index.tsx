import "./app/ui-core/theme/GlobalStyle/roboto.css";
import ReactDOM from "react-dom";
import { Provider } from "react-redux";
import { store } from "./redux";
import { ServicedApp, ThemeProviderConnected } from "./app";

ReactDOM.render(
  <Provider store={store}>
    <ThemeProviderConnected>
      <ServicedApp />
    </ThemeProviderConnected>
  </Provider>,
  document.getElementById("root")
);
