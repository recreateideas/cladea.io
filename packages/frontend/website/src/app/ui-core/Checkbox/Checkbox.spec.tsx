import { Checkbox } from './Checkbox';
import { Checkbox as CheckboxCore } from '@mui/material';
import { TestProvider } from 'src/tests';

const { mountWithProviderAndTheme } = TestProvider();
describe('<Checkbox/>', () => {
    it('should render', () => {
        const state = {
            common: {},
        };
        const root = mountWithProviderAndTheme(<Checkbox />, state);
        expect(root).toMatchSnapshot();
    });
    it('should render with the label', () => {
        const state = {
            common: {},
        };
        const root = mountWithProviderAndTheme(<Checkbox label="some-label" />, state);
        expect(root).toMatchSnapshot();
    });
    it('should be checked if checked prop is true', () => {
        const state = {
            common: {},
        };
        const root = mountWithProviderAndTheme(<Checkbox checked />, state);
        const checkbox = root.find(CheckboxCore).at(0);
        expect(checkbox.props().checked).toBe(true);
    });
    it('should be checked if value prop is true', () => {
        const state = {
            common: {},
        };
        const root = mountWithProviderAndTheme(<Checkbox value />, state);
        const checkbox = root.find(CheckboxCore).at(0);
        expect(checkbox.props().checked).toBe(true);
    });
});
