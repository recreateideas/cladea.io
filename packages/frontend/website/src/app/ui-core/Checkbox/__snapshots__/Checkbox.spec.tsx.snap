// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox/> should render 1`] = `
Array [
  .c4 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: fill 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
  font-size: 1.5rem;
}

.c5 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
}

.c0::-moz-focus-inner {
  border-style: none;
}

.c0.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

.c1 {
  padding: 9px;
  border-radius: 50%;
}

.c3 {
  cursor: inherit;
  position: absolute;
  opacity: 0;
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
  margin: 0;
  padding: 0;
  z-index: 1;
}

.c2 {
  color: #8697BF;
}

.c2:hover {
  background-color: rgba(18,216,247,0.04);
}

.c2.Mui-checked,
.c2.MuiCheckbox-indeterminate {
  color: #12D8F7;
}

.c2.Mui-disabled {
  color: rgba(0,0,0,0.26);
}

@media print {
  .c0 {
    color-adjust: exact;
  }
}

@media (hover:none) {
  .c2:hover {
    background-color: transparent;
  }
}

<div
    className=""
  >
    <span
      className="c0 MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root c1 MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root c2"
      onBlur={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={null}
    >
      <input
        checked={false}
        className="c3 PrivateSwitchBase-input"
        data-indeterminate={false}
        onChange={[Function]}
        type="checkbox"
      />
      <svg
        aria-hidden={true}
        className="c4 MuiSvgIcon-root MuiSvgIcon-fontSizeMedium"
        data-testid="CheckBoxOutlineBlankIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
        />
      </svg>
      <span
        className="c5 MuiTouchRipple-root"
      />
    </span>
  </div>,
  "",
]
`;

exports[`<Checkbox/> should render with the label 1`] = `
Array [
  .c5 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: fill 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
  font-size: 1.5rem;
}

.c6 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

.c1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
}

.c1::-moz-focus-inner {
  border-style: none;
}

.c1.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

.c7 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  -webkit-letter-spacing: 0.00938em;
  -moz-letter-spacing: 0.00938em;
  -ms-letter-spacing: 0.00938em;
  letter-spacing: 0.00938em;
}

.c2 {
  padding: 9px;
  border-radius: 50%;
}

.c4 {
  cursor: inherit;
  position: absolute;
  opacity: 0;
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
  margin: 0;
  padding: 0;
  z-index: 1;
}

.c3 {
  color: #8697BF;
}

.c3:hover {
  background-color: rgba(18,216,247,0.04);
}

.c3.Mui-checked,
.c3.MuiCheckbox-indeterminate {
  color: #12D8F7;
}

.c3.Mui-disabled {
  color: rgba(0,0,0,0.26);
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  vertical-align: middle;
  -webkit-tap-highlight-color: transparent;
  margin-left: -11px;
  margin-right: 16px;
}

.c0.Mui-disabled {
  cursor: default;
}

.c0 .MuiFormControlLabel-label.Mui-disabled {
  color: rgba(0,0,0,0.38);
}

@media print {
  .c1 {
    color-adjust: exact;
  }
}

@media (hover:none) {
  .c3:hover {
    background-color: transparent;
  }
}

<div
    className=""
  >
    <label
      className="c0 MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd"
    >
      <span
        className="c1 MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root c2 MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root c3"
        label="some-label"
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={null}
      >
        <input
          checked={false}
          className="c4 PrivateSwitchBase-input"
          data-indeterminate={false}
          onChange={[Function]}
          type="checkbox"
        />
        <svg
          aria-hidden={true}
          className="c5 MuiSvgIcon-root MuiSvgIcon-fontSizeMedium"
          data-testid="CheckBoxOutlineBlankIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
          />
        </svg>
        <span
          className="c6 MuiTouchRipple-root"
        />
      </span>
      <span
        className="c7 MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label"
      >
        some-label
      </span>
    </label>
  </div>,
  "",
]
`;
